ERCS/erc-7674.md

```diff
@@ -0,0 +1,58 @@
+---
+eip: 7674
+title: Temporary Approval Extension for ERC-20
+description: An interface for ERC-20 approvals via transient storage
+author: Xenia Shape (@byshape), Mikhail Melnik (@ZumZoom), Hadrien Croubois (@Amxx)
+discussions-to: https://ethereum-magicians.org/t/add-erc-7674-transient-approval-extension-for-erc-20/19521
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-04-02
+requires: 20, 1153
+---
+
+## Abstract
+
+This specification defines the minimum interface required to temporarily approve `ERC-20` tokens for spending within a single transaction.
+
+## Motivation
+
+`EIP-1153` allows to use a cheaper way to temporarily store allowances.
+
+## Specification
+
+The key words "MUST", "SHOULD", "MAY" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+Compliant contracts MUST implement 1 new function in addition to `ERC-20`:
+```solidity
+function temporaryApprove(address spender, uint256 value) public returns (bool success)
+```
+Call to `temporaryApprove(spender, value)` allows `spender` to withdraw within the same transaction from `msg.sender` multiple times, up to the `value` amount.
```

Comment:
I think this should be updated regardless.

Reply:
If we're talking about the same transaction, I think it's better to mention only `value` since `allowance(owner, spender)` can be spent (or not) later too.

Reply:
```suggestion
Call to `temporaryApprove(spender, value)` allows `spender` to withdraw within the same transaction from `msg.sender` multiple times, up to the `value` amount. This temporary allowance is to be considered in addition to the normal (persistent) ERC20 allowance, the total value than spender is able to spend during the transaction is thus capped by the sum of the temporary and the normal (pensistent) allowances.
```